src/java.base/share/classes/sun/security/ssl/SSLSessionContextImpl.java

```diff
@@ -170,6 +178,52 @@ public int getSessionCacheSize() {
         return cacheLimit;
     }
 
+    // Package-private, used only from SSLContextImpl::engineInit() to initialie currentKeyID.
+    void initCurrentKeyID(int keyID) {
+        currentKeyID = keyID;
+    }
+
+    // Package-private, used only from SSLContextImpl::getKey() to create a new session key.
+    int getCurrentKeyID() {
```

Comment:
This method could be removed if you want to move SSLContextImpl.getKey() into this class.

Reply:
Done.